pkg/location/location_test.go

```diff
@@ -107,18 +112,95 @@ func (s *LocationSuite) TearDownTest(c *C) {
 		err := s.root.Delete(ctx, s.testpath)
 		if err != nil {
 			c.Log("Cannot cleanup test directory: ", s.testpath)
-			return
+		}
+	}
+	if s.testMultipartPath != "" {
+		c.Assert(s.root, NotNil)
+		ctx := context.Background()
+		err := s.root.Delete(ctx, s.testMultipartPath)
+		if err != nil {
+			c.Log("Cannot cleanup test directory: ", s.testMultipartPath)
 		}
 	}
 }
 
 func (s *LocationSuite) TestWriteAndReadData(c *C) {
 	ctx := context.Background()
-	teststring := "test-content"
+	teststring := "test-content-check"
 	err := writeData(ctx, s.osType, s.profile, bytes.NewBufferString(teststring), s.testpath)
 	c.Check(err, IsNil)
 	buf := bytes.NewBuffer(nil)
 	err = readData(ctx, s.osType, s.profile, buf, s.testpath)
 	c.Check(err, IsNil)
 	c.Check(buf.String(), Equals, teststring)
 }
+
+func (s *LocationSuite) TestAzMultipartUpload(c *C) {
+	if s.osType != objectstore.ProviderTypeAzure {
+		c.Skip(fmt.Sprintf("Not applicable for location type %s", s.osType))
+	}
+
+	// Create dir if not exists
+	_, err := os.Stat(s.suiteDirPrefix)
+	if os.IsNotExist(err) {
+		err := os.MkdirAll(s.suiteDirPrefix, 0755)
+		c.Check(err, IsNil)
+	}
+	// Create test file
+	f, err := os.Create(s.testMultipartPath)
+	c.Check(err, IsNil)
+	defer f.Close()
+
+	ctx := context.Background()
+	for _, fileSize := range []int64{
+		0,                 // empty file
+		100 * 1024 * 1024, // 100M ie < buffSize
```

Comment:
Please test the boundary cases - 256MiB -1, 256MiB, 256MiB+1

Reply:
Sure